$NetBSD: patch-al,v 1.1 2007/12/22 23:12:04 ghen Exp $

If LDAP base contained variables, auth cache should have included
them in the cache key.

http://www.dovecot.org/list/dovecot-news/2007-December/000057.html

--- src/auth/passdb-ldap.c.orig	2007-12-11 19:52:08.000000000 +0100
+++ src/auth/passdb-ldap.c
@@ -519,7 +519,8 @@ passdb_ldap_preinit(struct auth_passdb *
 			  conn->set.auth_bind ? "password" : NULL);
 	module->module.cache_key =
 		auth_cache_parse_key(auth_passdb->auth->pool,
-				     conn->set.pass_filter);
+				     t_strconcat(conn->set.base,
+						 conn->set.pass_filter, NULL));
 	module->module.default_pass_scheme = conn->set.default_pass_scheme;
 	return &module->module;
 }
$NetBSD: patch-al,v 1.1 2007/12/22 23:12:04 ghen Exp $

--- src/auth/userdb-ldap.c.orig	2007-12-11 19:52:08.000000000 +0100
+++ src/auth/userdb-ldap.c
@@ -231,7 +231,8 @@ userdb_ldap_preinit(struct auth_userdb *
 			  conn->user_attr_map, default_attr_map, NULL);
 	module->module.cache_key =
 		auth_cache_parse_key(auth_userdb->auth->pool,
-				     conn->set.user_filter);
+				     t_strconcat(conn->set.base,
+						 conn->set.user_filter, NULL));
 	return &module->module;
 }
 
